docs/EN_US/ECLProgrammersGuide/PRG_Mods/PrGit_Source_Code.xml

```diff
@@ -0,0 +1,610 @@
+
+      <para>Using a single repository had some drawbacks. You could have
+      multiple copies of the same source code not really being in sync. You
+      could also have problems with incompatible changes, dependencies, or
+      clashes if there are modules with the same name. Using multiple
+      repositories solves these issues. Queries compiled from multiple
+      repositories, avoid problems with clashing symbols. It prevents clashes
+      and ensures that the dependencies between repositories are
+      versioned.</para>
+
+      <para>The HPCC Systems platform now allows each Git repository to be
+      treated as a separate independent package. Dependencies between the
+      repositories are specified in a package file which is checked into the
+      repository and versioned along with the ECL code. The package file
+      indicates what the dependencies are and which versions should be
+      used.</para>
+
+      <para>There are plans to add support for dependencies between multiple
+      repositories in a future release.</para>
+    </sect3>
+  </sect2>
+
+  <sect2>
+    <title>Using Git with HPCC</title>
+
+    <para>The --main syntax has been extended to allow compiling directly from
+    the repository.</para>
+
+    <para>Consider the following command :</para>
+
+    <programlisting>ecl run thor --main demo.main@https://github.com/gituser/gch-demo-d#version1 --server=... </programlisting>
+
+    <para>This command submits a query to Thor via ESP. It retrieves ECL code
+    from the 'version1' branch in the https://hithub.com/gituser/gch-demo-d
+    repository. Compiles the code in the demo/main.ecl file and then runs the
+    query on Thor. The checkout will be done on the remote ECLCC Server rather
+    than on the client machine.</para>
+
+    <sect3 id="RepoReferenceSyntax">
+      <title>Repository Reference Syntax</title>
+
+      <para>The syntax for the reference to the repository is as
+      follows:</para>
+
+      <programlisting>&lt;protocol:&gt;//&lt;urn&gt;/&lt;user&gt;/&lt;repository&gt;#version</programlisting>
+
+      <para>The protocol and urn can be omitted and a default will be used.
+      Such as in the following example:</para>
+
+      <programlisting>ecl run thor --main demo.main@gituser/gch-ecldemo-d#version1 --server=...</programlisting>
+
+      <para>This command also submits a query to Thor, retrieves ECL code from
+      the 'version1' branch in the gch-demo-d repository. Compiles the code in
+      the demo.main.ecl file and then runs the query on Thor.</para>
+    </sect3>
+
+    <sect3 id="HashVersionText">
+      <title>Version-Text</title>
+
+      <para>The version text that follows the hash (#) in the repository
+      reference can take any of the following forms:</para>
+
+      <itemizedlist>
+        <listitem>
+          <para>The name of a branch</para>
+        </listitem>
+
+        <listitem>
+          <para>The name of a tag</para>
```

Comment:
Note: Currently only lightweight tags are supported.  Annotated tags are not yet supported.